libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/TypeRep.hs

```diff
@@ -451,6 +453,50 @@ genpup (PPUPStateR (Alonzo _)) = arbitrary
 genpup (PPUPStateR (Babbage _)) = arbitrary
 genpup (PPUPStateR (Conway _)) = arbitrary -- FIXME when Conway is fully defined.
 
+-- ===========================
+-- QuickCheck shrinking
+
+shrinkRep :: Era era => Rep era t -> t -> [t]
+shrinkRep CoinR t = shrink t
+shrinkRep (_ :-> _) _ = []
+shrinkRep (MapR a b) t = shrinkMapBy Map.fromList Map.toList (shrinkRep $ ListR (PairR a b)) t
+shrinkRep (SetR a) t = shrinkMapBy Set.fromList Set.toList (shrinkRep $ ListR a) t
+shrinkRep (ListR a) t = shrinkList (shrinkRep a) t
+shrinkRep CredR t = shrink t
+shrinkRep PoolHashR t = shrink t
+shrinkRep WitHashR t = shrink t
+shrinkRep GenHashR t = shrink t
+shrinkRep GenDelegHashR t = shrink t
+shrinkRep PoolParamsR t = shrink t
+shrinkRep EpochR t = shrink t
+shrinkRep RationalR t = shrink t
+shrinkRep Word64R t = shrink t
+shrinkRep IntR t = shrink t
+shrinkRep NaturalR t = shrink t
+shrinkRep FloatR t = shrink t
+shrinkRep TxInR t = shrink t
+shrinkRep (ValueR _) _ = []
+shrinkRep (TxOutR _) _ = []
+shrinkRep (UTxOR _) _ = []
+shrinkRep (PParamsR _) _ = []
```

Comment:
Why are somethings using (shrink t) and others ([]).  How do you know? experience? or something else?

Reply:
We'd like them all to be `shrink t`. It's `[]` for the ones that don't have an `Arbitrary` instance. I'll add a todo.